src/Dialect/ONNX/IndexExpr.hpp

```diff
@@ -311,7 +311,7 @@ enum class IndexExprKind {
 // Data structure to hold all the IndexExpr in a given scope. A scope define
 // a scope during which each of the dynamic dimensions are defined and all of
 // the symbols hold constant in that scope.
-class IndexExprScope {
+class IndexExprScope final {
```

Comment:
Wanted to get rid of `virtual` on the destructor, so declaring this class `final`.